executors/rust/src/numberfmt.rs

```diff
@@ -71,14 +85,34 @@ pub fn run_numberformat_test(json_obj: &Value) -> Result<Value, String> {
     // !! A test. More options to consider!
     options.grouping_strategy = options::GroupingStrategy::Min2;
 
-    // Can this fail with invalid options?
-    let fdf = FixedDecimalFormatter::try_new_unstable(&provider, &data_locale, options)
-        .expect("Data should load successfully");
-
     // Returns error if parsing the number string fails.
     let input_num = input.parse::<FixedDecimal>().map_err(|e| e.to_string())?;
-    // TODO: Can this fail?
-    let result_string = fdf.format(&input_num);
+
+    let mut result_string = if is_compact {
+        let input_num = input.parse::<i64>().map_err(|e| e.to_string())?;
+        // We saw compact!
+        let cdf = if compact_type == "short" {
+            CompactDecimalFormatter::try_new_short_unstable(
+                &provider,
+                &data_locale,
+                Default::default(),
+            ).unwrap()  // ??  options);
+        } else {
+            CompactDecimalFormatter::try_new_long_unstable(
+                &provider,
+                &data_locale,
+                Default::default(),
+            ).unwrap()  // ??  options);
+        }
+        cdf.format_i64(input_num)
+    }
+    else {
+        let input_num = input.parse::<FixedDecimal>().map_err(|e| e.to_string())?;
```

Comment:
Note that we will need to set the minimum/maximum significant/fraction digits somewhere around here.